File: compiler/testData/objc/java/nil.m
#import "nil.h"

@implementation A
+ (id) getNilObject {
    return nil;
}
+ (A *) getNilA {
    return nil;
}
+ (Class) getNilClass {
    return Nil;
}
+ (void *) getNullPointer {
    return NULL;
}
+ (S *) getNullS {
    return NULL;
}


+ (BOOL) isNilObject: (id) object {
    return object == nil;
}
+ (BOOL) isNilA: (A *) a {
    return a == nil;
}
+ (BOOL) isNilClass: (Class) class {
    return class == Nil;
}
+ (BOOL) isNullPointer: (void *) pointer {
    return pointer == NULL;
}
+ (BOOL) isNullS: (S *) s {
    return s == NULL;
}
@end

File: compiler/testData/objc/java/nil.kt
package test

import objc.*

fun test(): String {
    val nil = A.getNilObject()
    if (nil != ObjCObject.NIL)
        return "Fail nil: $nil"

    val a = A.getNilA()
    if (a != ObjCObject.NIL)
        return "Fail a: $a"

    val Nil = A.getNilClass()
    if (Nil != ObjCClass.NIL)
        return "Fail Nil: $Nil"

    val nullPtr = A.getNullPointer()
    if (nullPtr != Pointer.NULL)
        return "Fail null: $nullPtr"

    val s = A.getNullS()
    if (s != Pointer.NULL)
        return "Fail s: $s"


    if (!A.isNilObject(ObjCObject.NIL))
        return "Fail isNilObject"

    /* TODO
    if (!A.isNilA(A(null)))
        return "Fail isNilA"
    */

    if (!A.isNilClass(ObjCClass.NIL))
        return "Fail isNilClass"

    if (!A.isNullPointer(Pointer.NULL))
        return "Fail isNullPointer"

    if (!A.isNullS(Pointer.NULL))
        return "Fail isNullS"

    return "OK"
}

fun main(args: Array<String>) {
    print(test())
}
